### PR TITLE
Add connect protocol handler to SQL command line service

### PR DESCRIPTION
**Pull request by kevcunnane**
- This already handles commandline which needs same logic
- Includes visual confirmation to avoid connecting by default to untrusted server
- If successful will connect & expand in Connections tree
- If failed, will show dialog with all presets
- Shows New Query by default, can add option for New Notebook instead in the future

Connection protocol is `azuredatastudio://connect?server=mysqldw.database.windows.net&database=db1&user=myusername&aad=true`
Same args as commandline, namely:
- server
- database
- user
- aad
- integrated

I have also added `provider` arg to allow non-MSSQL providers. In the future, we should update logic to prompt for install of correct extension for that provider if it's missing.

ConnectProtocol

### CODE REVIEW

File: src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts [hunk 1]
```diff
@@ -26,8 +27,24 @@ import { INotificationService } from 'vs/platform/notification/common/notificati
...
+import { IURLService, IURLHandler } from 'vs/platform/url/common/url';
+import { parse } from 'vs/base/common/marshalling';
+import { getErrorMessage } from 'vs/base/common/errors';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
 
-export class CommandLineWorkbenchContribution implements IWorkbenchContribution {
+const connectAuthority = 'connect';
+
+interface SqlArgs {
+	_?: string[];
```
Comment (corivera) [inline src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts]: What is this one for?

Comment (kevcunnane) [reply src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts]: It's needed by command line path, used to get all unnamed args (which are then treated as files). Using this interface let me avoid doing dummy inputs for all the commandline args, none of which this service uses

File: src/sql/workbench/parts/commandLine/test/electron-browser/commandLine.test.ts [hunk 2]
```diff
@@ -115,6 +115,8 @@ suite('commandLineService tests', () => {
 			configurationService,
 			undefined,
 			logService,
+			undefined,
+			undefined,
 			undefined
 		);
 	}
```
Comment (shueybubbles) [inline src/sql/workbench/parts/commandLine/test/electron-browser/commandLine.test.ts]: no tests?

File: src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts [hunk 3]
```diff
@@ -26,8 +27,25 @@ import { INotificationService } from 'vs/platform/notification/common/notificati
 import { IWorkbenchContribution } from 'vs/workbench/common/contributions';
 import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
 import { openNewQuery } from 'sql/workbench/parts/query/browser/queryActions';
+import { IURLService, IURLHandler } from 'vs/platform/url/common/url';
+import { parse } from 'vs/base/common/marshalling';
```
Comment (anthonydresser) [inline src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts]: unused import

File: src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts [hunk 4]
```diff
@@ -130,6 +153,48 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution
 		}
 	}
 
+	async handleURL(uri: URI): Promise<boolean> {
```
Comment (anthonydresser) [inline src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts]: Unit tests? Seems like it should be simple to do some basic tests with urls we know we will get from callers.

File: src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts [hunk 5]
```diff
@@ -26,8 +27,24 @@ import { INotificationService } from 'vs/platform/notification/common/notificati
...
+
+interface SqlArgs {
+	_?: string[];
+	aad?: boolean;
+	database?: string;
+	integrated?: boolean;
+	server?: string;
+	user?: string;
+	command?: string;
+	provider?: string;
```
Comment (shueybubbles) [inline src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts]: should this be connectionProvider to disambiguate, should we ever add it to ParsedArgs?

Comment (kevcunnane) [reply src/sql/workbench/parts/commandLine/electron-browser/commandLine.ts]: Yeah I can rename - trying to keep it short is the only thing. Which do you think is better (short vs. very clear)?

File: src/sql/workbench/parts/commandLine/test/electron-browser/commandLine.test.ts (hunk 2 above)
Comment (kevcunnane) [reply src/sql/workbench/parts/commandLine/test/electron-browser/commandLine.test.ts]: Will add

**Pull request merged**